kustomize/commands/edit/fix/convert_test.go

```diff
@@ -1143,3 +1143,76 @@ metadata:
     a.b.c: SOME_SECRET_NAME_PLACEHOLDER
 `, string(content))
 }
+
+func TestFixVarsWithPatch(t *testing.T) {
+	kustomization := []byte(`
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+patchesStrategicMerge:
+  - patch.yaml
+
+vars:
+  - name: CERTIFICATE_NAMESPACE
+    objref:
+      name: system
+    fieldref:
+      fieldpath: metadata.namespace
+`)
+	patch := []byte(`
+apiVersion: apps/v1
+kapiVersion: apps/v1
```

Comment:
I believe this might be a typo (repeated on lines 1206-1207).

Reply:
Removed the second line.